spec/controllers/moderator_applications_controller_spec.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+# The application ID to use in requests.
+def application_id
+  # @application.id will always be set unless :does_not_exist is set for the example,
+  # in which case we specifically want to use an id that does not exist.
+  @application.id || 192_139
+end
+
+def it_handles_nonexistent_application
+  it 'handles a non-existent application', :does_not_exist do
+    expect_not_found
+  end
+end
+
+def it_requires_that_applications_are_open
+  it 'requires that applications are open', :applications_closed do
+    expect_unauthorized
+  end
+end
+
+def it_requires_login
+  it 'requires login', :no_login do
+    expect_unauthenticated
+  end
+end
+
+def it_requires_correct_login
+  it_requires_login
+
+  it 'requires correct login', :incorrect_login do
+    expect_unauthorized
+  end
+end
+
+# The show and edit routes both have identical tests, which I have factored out here.
+def show_and_edit_tests(route)
```

Comment:
Okay, but *why* are the tests identical? Also, this interrupts the flow of reading the code enough I'm not sure it was even worth factoring out, readability-wise.